src/components/signin/identity/QuickInfo.js

```diff
@@ -0,0 +1,225 @@
+import React, { Component } from 'react'
+import IdentityHoc from './IdentityHoc'
+import Grid from '@material-ui/core/Grid'
+import TextField from '@material-ui/core/TextField'
+import Translate from 'components/translate/Translate'
+import Checkbox from '@material-ui/core/Checkbox'
+import FormControlLabel from '@material-ui/core/FormControlLabel'
+import FormHelperText from '@material-ui/core/FormHelperText'
+import FormControl from '@material-ui/core/FormControl'
+import { withStyles } from '@material-ui/core/styles'
+import { styles } from './styles'
+import { validEmail, validPassword } from 'helpers/validators'
+
+class GrantInfo extends Component {
+	componentDidMount() {
+		const {
+			email,
+			emailCheck,
+			password,
+			passwordCheck,
+			tosCheck,
+		} = this.props.identity
+
+		this.validateEmail(email, false)
+		this.validateEmailCheck(emailCheck, false)
+		this.validatePassword(password, false)
+		this.validatePasswordCheck(passwordCheck, false)
+		this.validateTOS(tosCheck, false)
+	}
+
+	validateEmail(email, dirty) {
+		const isValid = validEmail(email)
+		this.props.validate('email', {
+			isValid: isValid,
+			err: { msg: 'ERR_EMAIL' },
+			dirty: dirty,
+		})
+	}
+
+	validateEmailCheck(emailCheck, email, dirty) {
+		const isValid = !!emailCheck && !!email && emailCheck === email
+		this.props.validate('emailCheck', {
+			isValid: isValid,
+			err: { msg: 'ERR_EMAIL_CHECK' },
+			dirty: dirty,
+		})
+	}
+
+	validatePassword(password, dirty) {
+		const isValid = validPassword(password)
+		this.props.validate('password', {
+			isValid: isValid,
+			err: { msg: 'ERR_PASSWORD' },
+			dirty: dirty,
```

Comment:
why not just `dirty` or just `isValid` (ES6)

Reply:
if it's not `dirty` it will not show the error msg, but will not be valid to continue. This is used to not show the errors when you open the step

Reply:
There is more work on quick account though. (create and recover)

Reply:
ES6 ... you right